src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,40 +205,107 @@ function updateStatsCheckBox() {
     $('input[id=statsShownCheckBox]').val($('#show-stats-checkbox').is(':checked'));
 }
 
-function toggleCollapse(e, pans) {
-    var expand = 'Expand';
-    var collapse = 'Collapse';
-    var panels = pans || $('div.panel-collapse');
-
-    if ($(e).html().trim().startsWith(expand)) {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
-            }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+/**
+ * Expands or collapses all panels when collapse/expand panels button is clicked.
+ * @param {DOM} expandCollapseButton - The button that was clicked to invoke {@code #expandOrCollapsePanels}
+ * @param {DOM} panels - The panels to be expanded/collapsed. Not defined if {@param expandCollapseButton}
+ *         is collapse panels button.
+ */
+function expandOrCollapsePanels(expandCollapseButton, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    // {@param panels} is not defined when @param expandCollapseButton} is collapse panels button. We
+    // need to define corresponding {@code targetPanels}.
+    var targetPanels = panels || $('div.panel-collapse');
```

Comment:
I thought there is only one button that is being used to expand and collapse? What are other button?
![image](https://cloud.githubusercontent.com/assets/1673303/24251761/3a9427e6-1016-11e7-8784-99e6b6d505a9.png)